crates/utils/src/utils/validation.rs

```diff
@@ -17,32 +17,72 @@ static CLEAN_URL_PARAMS_REGEX: Lazy<Regex> = Lazy::new(|| {
   Regex::new(r"^utm_source|utm_medium|utm_campaign|utm_term|utm_content|gclid|gclsrc|dclid|fbclid$")
     .expect("compile regex")
 });
+const BODY_MAX_LENGTH: usize = 10000;
+const BIO_MAX_LENGTH: usize = 300;
 
 fn has_newline(name: &str) -> bool {
   name.contains('\n')
 }
 
-pub fn is_valid_actor_name(name: &str, actor_name_max_length: usize) -> bool {
-  name.chars().count() <= actor_name_max_length
+pub fn is_valid_actor_name(name: &str, actor_name_max_length: usize) -> Result<(), LemmyError> {
+  let check = name.chars().count() <= actor_name_max_length
     && VALID_ACTOR_NAME_REGEX.is_match(name)
-    && !has_newline(name)
+    && !has_newline(name);
+  if !check {
+    Err(LemmyError::from_message("invalid_name"))
+  } else {
+    Ok(())
+  }
 }
 
 // Can't do a regex here, reverse lookarounds not supported
-pub fn is_valid_display_name(name: &str, actor_name_max_length: usize) -> bool {
-  !name.starts_with('@')
+pub fn is_valid_display_name(name: &str, actor_name_max_length: usize) -> Result<(), LemmyError> {
+  let check = !name.starts_with('@')
     && !name.starts_with('\u{200b}')
     && name.chars().count() >= 3
     && name.chars().count() <= actor_name_max_length
-    && !has_newline(name)
+    && !has_newline(name);
+  if !check {
+    Err(LemmyError::from_message("invalid_username"))
+  } else {
+    Ok(())
+  }
+}
+
+pub fn is_valid_matrix_id(matrix_id: &str) -> Result<(), LemmyError> {
+  let check = VALID_MATRIX_ID_REGEX.is_match(matrix_id) && !has_newline(matrix_id);
+  if !check {
+    Err(LemmyError::from_message("invalid_matrix_id"))
+  } else {
+    Ok(())
+  }
 }
 
-pub fn is_valid_matrix_id(matrix_id: &str) -> bool {
-  VALID_MATRIX_ID_REGEX.is_match(matrix_id) && !has_newline(matrix_id)
+pub fn is_valid_post_title(title: &str) -> Result<(), LemmyError> {
+  let check = VALID_POST_TITLE_REGEX.is_match(title) && !has_newline(title);
+  if !check {
+    Err(LemmyError::from_message("invalid_post_title"))
+  } else {
+    Ok(())
+  }
 }
 
-pub fn is_valid_post_title(title: &str) -> bool {
-  VALID_POST_TITLE_REGEX.is_match(title) && !has_newline(title)
+pub fn is_valid_body_field(body: &str) -> Result<(), LemmyError> {
```

Comment:
Would call this is_valid_post_body() for clarity.

Reply:
Nevermind it seems to be used both for posts and comments. Would then add a comment mentioning that.